### ISSUE TITLE
No errors/warning for usage of moved value with structs

### ISSUE THREAD

**Issue by synepis** [rust-lang/rust#55653]
The Rust compiler doesn't complain upon trying to use a moved value.
Note that there seems to be no side effects, but it should still be disallowed or at least warned against.

Example (try it on [Playgound](https://play.rust-lang.org/?version=stable&mode=debug&edition=2015&gist=bdfd5c4237cc5120c86a16c5121b0290)):
```rust
#[derive(Debug)]
struct Point { x: i32, y: i32 }

fn main() {
    let mut p1 = Point { x: 1, y: 2 };
    println!("p1: {:?}", p1);    // p1: Point { x: 1, y: 2 }
    p1.x = -1;
    println!("p1: {:?}", p1);    //  p1: Point { x: -1, y: 2 }

    let mut p2 = p1;
    println!("p2: {:?}", p2);    // p2: Point { x: -1, y: 2 }
    p2.x = 3;
    // println!("p1: {:?}", p1); // Compiler error: "p1 was already moved!"
    p1.x = -11;                  // Compiles but what does it do?
    println!("p2: {:?}", p2);    // p2: Point { x: 3, y: 2
}
```
I would expect  `p1.x = -11;` not to compile.
``` 
$ rustc --version --verbose
rustc 1.28.0
binary: rustc
commit-hash: unknown
commit-date: unknown
host: x86_64-unknown-linux-gnu
release: 1.28.0
```

**Comment by sfackler** [comment]
Duplicate of #21232.
